libsplinter/src/matrix.rs

```diff
@@ -12,54 +12,120 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Traits that abstract specific functionality of connection group.
+//!
+//! A component that acts as a connection group peforms the following operations
+//!   - add: Adds a connection to the underlying connection group
+//!   - remove: Removes a connection from the connection group
+//!   - send: Send a message over a specific connection
+//!   - recv: Receive a message from any ready connection in the connection group
+//!
+//! The following Matrix traits abstract out these different operations. For example an
+//! and networking component may only be interested in send and receiving messages. That component
```

Comment:
I assume this should be "For example, a networking component..."? Also s/send/sending/